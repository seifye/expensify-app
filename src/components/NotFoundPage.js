// import React for using JSX
import React from 'react'

/* Link can be used to do client side routing behind the scenes without a full page refresh to navigate
the different pages in our app */
import { Link } from 'react-router-dom'



const NotFoundPage = () => (
    <div>
        This is from 404 page --- <Link to='/'> Go Home </Link>
    </div>
)


export default NotFoundPage