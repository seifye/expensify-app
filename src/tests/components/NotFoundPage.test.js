// import React for using JSX
import React from 'react'

/* There are two main ways to test react components (Shallow rendering and full DOM rendering). 
Shallow rendering renders only a given component while full DOM rendering renders child components as
well */

// Importing the shallow renderer from enzyme library
import { shallow } from 'enzyme'

// import other js files => use
import NotFoundPage from '../../components/NotFoundPage'



// Testing React Component
// This is a static component. It doesn't have any state, it doesn't take in any props
test('should render the NotFoundPage component correctly', () => {
    const wrapper = shallow(<NotFoundPage />)
    expect(wrapper).toMatchSnapshot()
})