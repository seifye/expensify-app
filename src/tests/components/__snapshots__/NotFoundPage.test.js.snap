// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the NotFoundPage component correctly 1`] = `
<div>
  This is from 404 page --- 
  <Link
    replace={false}
    to="/"
  >
     Go Home 
  </Link>
</div>
`;
